stake-pool/program/tests/helpers/mod.rs

```diff
@@ -0,0 +1,732 @@
+}
+
+pub async fn create_blank_stake_account(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    stake: &Keypair,
+) -> u64 {
+    let rent = banks_client.get_rent().await.unwrap();
+    let lamports = rent.minimum_balance(std::mem::size_of::<stake::StakeState>()) + 1;
+
+    let mut transaction = Transaction::new_with_payer(
+        &[system_instruction::create_account(
+            &payer.pubkey(),
+            &stake.pubkey(),
+            lamports,
+            std::mem::size_of::<stake::StakeState>() as u64,
+            &stake::id(),
+        )],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[payer, stake], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+
+    lamports
+}
+
+pub async fn create_stake_account(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    stake_pool: &Pubkey,
+    stake_account: &Pubkey,
+    validator: &Pubkey,
+    stake_authority: &Pubkey,
+    withdraw_authority: &Pubkey,
+) {
+    let mut transaction = Transaction::new_with_payer(
+        &[
+            instruction::create_validator_stake_account(
+                &id(),
+                &stake_pool,
+                &payer.pubkey(),
+                &stake_account,
+                &validator,
+                &stake_authority,
+                &withdraw_authority,
+                &solana_program::system_program::id(),
+                &stake::id(),
+            )
+            .unwrap(),
+            system_instruction::transfer(&payer.pubkey(), &stake_account, TEST_STAKE_AMOUNT),
+        ],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[payer], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+}
+
+pub async fn delegate_stake_account(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    stake: &Pubkey,
+    authorized: &Keypair,
+    vote: &Pubkey,
+) {
+    let mut transaction = Transaction::new_with_payer(
+        &[stake::delegate_stake(&stake, &authorized.pubkey(), &vote)],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[payer, authorized], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+}
+
+pub async fn authorize_stake_account(
+    banks_client: &mut BanksClient,
+    payer: &Keypair,
+    recent_blockhash: &Hash,
+    stake: &Pubkey,
+    authorized: &Keypair,
+    new_authorized: &Pubkey,
+    stake_authorize: stake::StakeAuthorize,
+) {
+    let mut transaction = Transaction::new_with_payer(
+        &[stake::authorize(
+            &stake,
+            &authorized.pubkey(),
+            &new_authorized,
+            stake_authorize,
+        )],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[payer, authorized], *recent_blockhash);
+    banks_client.process_transaction(transaction).await.unwrap();
+}
+
+pub struct StakeAccount {
```

Comment:
If you do not mind, I'll leave it to the next PR, now we're doing a bunch of negative tests which have a lot of changes in the test module, so it would be cleaner to do this after the tests are done.

Reply:
No problem